### PR TITLE
Release: 0.4.0

### LINKED ISSUE #21: Flexible config
It would be nice if you could pass arbitrary configs to the config file. You may want to publish to S3 or do something else that requires config beyond the basics the cookbook currently has. Perhaps the config should be driven by a resource that could take in a config hash.

### PR DESCRIPTION
**Pull request by YpNo**
Release: 0.4.0
- Big refacto
- Update resource with latest usage of aptly
- Create aptly_serve and aptly_api_serve resource
- Create unit tests
- Enhance integration test
- Use Circle CI

I think this release fix #21 #25 and #32 issues.

### CODE REVIEW

**Comment by YpNo** [comment]
Ping @damacus 
https://circleci.com/gh/YpNo/aptly/58?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
https://circleci.com/gh/YpNo/aptly/57?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by damacus** [comment]
Can you rebase on master? The CircleCI config has been update to use orbs in master

**Comment by YpNo** [comment]
@damacus, I don't know if I understand. I use orbs too in my config. I took postgresql cookbook as example (and I am up to date with master).

**Comment by YpNo** [comment]
Ping @damacus

**Comment by YpNo** [comment]
Ping @damacus

**Comment by damacus** [comment]
@YpNo this seems like a LOT of changes for release? See comments above

File: .kitchen.yml [hunk 1]
```diff
@@ -18,5 +19,7 @@ platforms:
 
 suites:
   - name: default
-    run_list:
-      - recipe[test::default]
+    verifier:
```
Comment (damacus) [inline .kitchen.yml]: This is the default value and can be removed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
 
 This file is used to list changes made in each version of the aptly cookbook.
 
+## 0.4.0
+
+- Big refacto
```
Comment (damacus) [inline CHANGELOG.md]: Typo here

Comment (YpNo) [reply CHANGELOG.md]: Hmm what Typo ? I did same as 0.3.0 release

File: .kitchen.yml (hunk 1 above)
Comment (YpNo) [reply .kitchen.yml]: ok

**Comment by YpNo** [comment]
Close to Open a new one to set a 1.0.0 release

**Pull request closed without merging**